cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/submit/MasterEnvironmentSparkSubmitter.java

```diff
@@ -140,6 +141,16 @@ protected void onCompleted(boolean succeeded) {
     }
   }
 
+  @Override
+  protected boolean hasSparkDriverSucceeded() {
+    try {
+      return generateOrGetSparkConfig().getStatusFuture().get(30, TimeUnit.SECONDS);
```

Comment:
does 30 sec timeout mean that driver pod will need to be in succeeded state within 30 seconds?

Reply:
increased timeout to be 10 mins